### PR TITLE
ENG-11061 Migration for deployment and alert field

### PR DESCRIPTION
**Pull request by qv-pat**
Added dropdown for datasources
Refactored add alert backend logic

### CODE REVIEW

File: superset/connectors/sqla/views.py [hunk 1]
```diff
@@ -411,6 +409,36 @@ def _delete(self, pk):
...
+        query = 'SHOW CATALOGS'
+        try:
+            results = presto_connection.execute(query)
+            for row in results:
+                deployment = row['Catalog']
+                if '_mysql' in deployment:
+                    deployment = deployment.split('_mysql')[0]
+                deployments.append(deployment)
+        except:
+            deployments = [
```
Comment (yuguang) [inline superset/connectors/sqla/views.py:429]: This is a list of dictionaries, not a list of strings. The above try statement would not be compatible

File: superset/connectors/sqla/views.py [hunk 2]
```diff
@@ -411,6 +409,36 @@ def _delete(self, pk):
         db.session.commit()
         self.update_redirect()
 
+    def get_deployment_names(self):
+        from sqlalchemy import create_engine
+        deployments = []
+        presto_engine = create_engine('presto://localhost:3894')
```
Comment (yuguang) [inline superset/connectors/sqla/views.py]: This can be read from the suprset_config

File: superset/migrations/versions/e201cd43a98f_.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+# revision identifiers, used by Alembic.
+revision = 'e201cd43a98f'
+down_revision = '3ba5164aa5dc'
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
```
Comment (yuguang) [inline superset/migrations/versions/e201cd43a98f_.py:17]: A migration for the existing entries needs to be added

**Review by yuguang — CHANGES_REQUESTED** [review]
Alert field still needs migrations and a UI

File: superset/connectors/sqla/views.py [hunk 4]
```diff
@@ -411,6 +409,36 @@ def _delete(self, pk):
...
+        presto_connection = presto_engine.connect()
+        query = 'SHOW CATALOGS'
+        try:
+            results = presto_connection.execute(query)
```
Comment (yuguang) [inline superset/connectors/sqla/views.py:420]: Remember to also add an option for "all" deployments

File: superset/connectors/sqla/views.py (hunk 1 above)
Comment (qv-pat) [reply superset/connectors/sqla/views.py:429]: The front end is expecting a list of dictionaries. I'll changed the above logic to do the same.

File: superset/connectors/sqla/views.py (hunk 2 above)
Comment (qv-pat) [reply superset/connectors/sqla/views.py]: what would I import it as?

File: superset/migrations/versions/e201cd43a98f_.py (hunk 3 above)
Comment (qv-pat) [reply superset/migrations/versions/e201cd43a98f_.py:17]: From my understanding this adds a column to the existing entries as well

File: superset/connectors/sqla/views.py (hunk 1 above)
Comment (qv-pat) [reply superset/connectors/sqla/views.py:429]: What kind of UI behavior do we want for alert field. Is it a dropdown as well? If so what are the options?

File: superset/connectors/sqla/views.py (hunk 2 above)
Comment (yuguang) [reply superset/connectors/sqla/views.py]: Call it PRESTO_SQLALCHEMY_URI

**Review by yuguang — APPROVED** [review]

**Pull request merged**